### PR TITLE
Add many-to-many relationship between patients and parents

### PR DESCRIPTION
**Pull request by thomasleese**
This follows up from #1736 and #1737 to finish the migration to have a many-to-many relationship between patients and parents. This removes the relationship fields from parents, and instead creates `ParentRelationship` records. The behaviour of the service should be the same before and after this change. See individual commits for more details.

### CODE REVIEW

File: db/migrate/20240912193815_remove_relationship_from_parents.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+class RemoveRelationshipFromParents < ActiveRecord::Migration[7.2]
+  def change
+    change_table :parents, bulk: true do |t|
+      t.remove :relationship, type: :integer
+      t.remove :relationship_other, type: :string
```
Comment (tvararu) [inline db/migrate/20240912193815_remove_relationship_from_parents.rb:7]: These are harmless but I don't think they do anything:

```suggestion
      t.remove :relationship
      t.remove :relationship_other
```

File: spec/factories/parents.rb [hunk 2]
```diff
@@ -11,32 +11,19 @@
...
 #  recorded_at                  :datetime
-#  relationship                 :integer
-#  relationship_other           :string
 #  created_at                   :datetime         not null
 #  updated_at                   :datetime         not null
 #
 FactoryBot.define do
   factory :parent do
     transient do
-      random { Random.new }
```
Comment (tvararu) [inline spec/factories/parents.rb:22]: Think we had a reason to allow passing in a transient `random:` attribute...

**Review by tvararu — APPROVED** [review]

File: spec/factories/parents.rb (hunk 2 above)
Comment (thomasleese) [reply spec/factories/parents.rb:22]: I did wonder about that, but then I couldn't see it was being used anywhere so I figured we could put it back if we needed to.

File: db/migrate/20240912193815_remove_relationship_from_parents.rb (hunk 1 above)
Comment (tvararu) [reply db/migrate/20240912193815_remove_relationship_from_parents.rb:7]: Oh fair

**Pull request merged**